File: gain_control.py
# --------------------------------------------------------------------
# Prototype gain control
# Capstone I
# --------------------------------------------------------------------
# NOTES:
"""
  -Volume and Pan can be implemented very easy in the same function.
  -
"""
# --------------------------------------------------------------------


# --------------------------------------------------------------------
# GAIN CONTROL - 
# --------------------------------------------------------------------
# GetTrack(int project number, int track number)
selected_track = RPR_GetTrack(0, 0)# Get track number(0 = 1st track)
vol = RPR_GetTrackUIVolPan(selected_track, 0, 0)[0] # Get volume of track

# Set track volume to half of current volume
RPR_SetMediaTrackInfo_Value(selected_track, "D_VOL", vol*0.5)

# RPR_SetMediaTrackInfo_Value(selected_track, "D_VOL", vol + hand_loc)

File: ReaScript/gain_control.lua
-- ------------------------------------------
-- Protoype Gain Control and Track selection
-- Captsone I
-- ------------------------------------------
-- NOTES: 
-- Vol & Pan can be changed in same function
-- 
-- ------------------------------------------
-- GAIN CONTROL
-- ------------------------------------------
-- Select track (0 = first num)
track = reaper.GetTrack(0, 2) --(project, track)
-- reaper.ShowConsoleMsg(tostring(track))

-- Get (retval, vol, pan)
ok, volume, pan = reaper.GetTrackUIVolPan(track, 0, 0)

--Set track 1 volume to half of current volume
reaper.SetMediaTrackInfo_Value(track, "D_VOL", volume*0.5)

if volume < 1 then
  reaper.SetMediaTrackInfo_Value(track, "D_VOL", 4) --12.04 dB
end

-- Possible implementation
-- RPR_SetMediaTrackInfo_Value(track, "D_VOL", volume + file_name.hand_location)
